### PR TITLE
Fix for the internal server error reported in #178

### PR DESCRIPTION
**Pull request by githubDante**
Additionally the vote will use the IP address from the request if the form
is not _Anonymous_ and the user is not logged in. See #179

### CODE REVIEW

**Comment by JanDragon** [comment]
Hey,

thanks for your first PR. Code looks good and is working for me. Please add a DCO and we are ready. More about DCO? Check out this link: https://stackoverflow.com/questions/1962094/what-is-the-sign-off-feature-in-git-for

**Comment by githubDante** [comment]
Fixed. Thanks for the advice.

File: lib/Controller/PageController.php [hunk 1]
```diff
@@ -274,7 +274,11 @@ public function gotoForm($hash) {
 		$participants = $this->voteMapper->findParticipantsByForm($form->getId());		
 
 		try {
-			$notification = $this->notificationMapper->findByUserAndForm($form->getId(), $this->userId);
+			if($this->userId === null) {
+				$notification = $this->notificationMapper->findByUserAndForm($form->getId(), '');
+			} else {
+				$notification = $this->notificationMapper->findByUserAndForm($form->getId(), $this->userId);
+			}
```
Comment (jotoeri) [inline lib/Controller/PageController.php]: This part should suffice to solve the error itself.

File: lib/Controller/PageController.php [hunk 2]
```diff
@@ -394,6 +398,9 @@ public function insertVote($id, $userId, $answers, $questions) {
 				if($form->getIsAnonymous()){
 						$vote->setUserId($anonID);
 				}else{
+						if ($userId === '') {
+							$userId = $this->request->getRemoteAddress();
+						}
```
Comment (jotoeri) [inline lib/Controller/PageController.php]: Why not using the already existing anonymous ID?

Comment (githubDante) [reply lib/Controller/PageController.php]: Because this only applies when the form is not marked as anonymous.

Comment (jotoeri) [reply lib/Controller/PageController.php]: Sure. But does the ip of an unidentified user contain helpful information? If i wanted to know who answered, i would make the form only visible to registered users...?!

Comment (jotoeri) [reply lib/Controller/PageController.php]: ......in fact, its not that big thing. Both works. ;)

**Comment by jotoeri** [comment]
Btw.: If you just amend an empty commit, the Travis-check should run again. Seems like the failing was not due to your code, but a wget() failed within this moment.

File: lib/Controller/PageController.php (hunk 2 above)
Comment (githubDante) [reply lib/Controller/PageController.php]: I didn't want to change anything in the current implementation, just to add on top of it. :)

Comment (jotoeri) [reply lib/Controller/PageController.php]: I'd say just use a hash of the ip, as @oscaropenness proposed on #179. Then push again and i would give it a go... ;)

Comment (jotoeri) [reply lib/Controller/PageController.php]: Ah, and maybe better place it in line 376 below the anonID. Then it also applies to Checkbox-Questions.

**Comment by jotoeri** [comment]
I think this PR can be closed, as the main-topic (internal server error) became obsolete with #235 .
If you continue working on, just open a new PR for the IP-Address-Topic.

**Pull request closed without merging**